spinnman/spalloc/spalloc_client.py

```diff
@@ -637,7 +715,7 @@ def _receive(self, timeout=None) -> bytes:
 
     def _is_ready_to_receive(self, timeout=0) -> bool:
         # If we already have a message or the queue peek succeeds, return now
-        if self.__current_msg is not None or self.__msgs.not_empty:
+        if self.__current_msg is not None or self.__msgs.qsize():
```

Comment:
Better as `not self.__msgs.empty()`, if my reading of the docs is right. We don't care how many items are in the queue, just whether there is anything.

Reply:
Fixed